### PR TITLE
Add show_private_test_case_results field to ora block

### PR DESCRIPTION
**Pull request by HussainTaj-arbisoft**
## Description

This PR adds a `show_private_test_case_results` boolean field to ORA Block. We will use this field in the future, to show the user results of the private test cases. Currently, the user can only see the results of the sample test cases.

Screenshot 2021-07-14 at 7 00 58 PM

## Ticket
[LT-221](https://arbisoft-assessment.atlassian.net/browse/LT-221?atlOrigin=eyJpIjoiZjdlMTFmYzJmZWYzNDBhMzg4ODYxZGM1ZDEzMzM0MTgiLCJwIjoiaiJ9)

## Testing
Tested on local machine.

### CODE REVIEW

File: openassessment/templates/openassessmentblock/edit/oa_edit.html [hunk 1]
```diff
@@ -65,6 +65,22 @@ <h2 class="openassessment_alert_title">{% trans "Rubric Change Impacts Settings
...
                     </p>
                 </li>
+                <li class="field comp-setting-entry">
+                    <div class="wrapper-comp-setting">
+                        <label for="openassessment_show_private_test_case_results_editor" class="setting-label">
+                            {% trans "Show Private Test Case Results"%}
+                        </label>
+                        <select id="openassessment_show_private_test_case_results_editor" class="input setting-input">
+                            <option value="0">{% trans "False"%}</option>
+                            <option value="1" {% if show_private_test_case_results %} selected="true" {% endif %}>
```
Comment (DawoudSheraz) [inline openassessment/templates/openassessmentblock/edit/oa_edit.html:75]: what is this if condition achieving?

File: openassessment/xblock/openassessmentblock.py [hunk 2]
```diff
@@ -255,6 +255,12 @@ class OpenAssessmentBlock(MessageMixin,
         help="List of strings relavent to the content."
     )
 
+    show_private_test_case_results = Boolean(
+        default=False,
+        scope=Scope.content,
+        help="Indicates whether or not to show private test case results (passed or failed only, does not show values)."
```
Comment (DawoudSheraz) [inline openassessment/xblock/openassessmentblock.py:261]: That is incorrect. For now, we will show both the values and results.

File: openassessment/templates/openassessmentblock/edit/oa_edit.html (hunk 1 above)
Comment (HussainTaj-arbisoft) [reply openassessment/templates/openassessmentblock/edit/oa_edit.html:75]: It is reflecting the `show_private_test_case_results` value inside the block on the frontend.

It sets the default selected option to "True" if `show_private_test_case_results` is true in the block. Without this, we will always see "False" selected whenever the settings HTML renders.

**Review by zainab-amir — APPROVED** [review]

File: openassessment/xblock/static/js/src/studio/oa_edit_settings.js [hunk 3]
```diff
@@ -290,6 +290,27 @@ OpenAssessment.EditSettingsView.prototype = {
...
+        boolean
+    **/
+    showPrivateTestCaseResultsEnabled: function(isEnabled) {
+        var sel = $('#openassessment_show_private_test_case_results_editor', this.settingsElement);
+        if (isEnabled !== undefined) {
+            if (isEnabled) {
+                sel.val(1);
+            } else {
+                sel.val(0);
+            }
```
Comment (DawoudSheraz) [inline openassessment/xblock/static/js/src/studio/oa_edit_settings.js:309]: you can simplify this. Just convert isEnable to 1 or 0(bool) and set that value

**Review by DawoudSheraz — APPROVED** [review]
one nit to address

**Pull request merged**